meerk40t/core/elements.py

```diff
@@ -3258,21 +3272,562 @@ def element_grid(
             y = y.value(ppi=1000, relative_length=height)
             if isinstance(x, Length) or isinstance(y, Length):
                 raise SyntaxError
-            y_pos = 0
+            if origin is None:
+                origin = (1, 1)
+            cx, cy = origin
             data_out = list(data)
+            if cx is None:
+                cx = 1
+            if cy is None:
+                cy = 1
+            # Tell whether original is at the left / middle / or right
+            start_x = -1 * x * (cx - 1)
+            start_y = -1 * y * (cy - 1)
+            y_pos = start_y
             for j in range(r):
-                x_pos = 0
+                x_pos = start_x
                 for k in range(c):
-                    if j != 0 or k != 0:
+                    if j != (cy - 1) or k != (cx - 1):
                         add_elem = list(map(copy, data))
                         for e in add_elem:
                             e *= "translate(%f, %f)" % (x_pos, y_pos)
                         self.add_elems(add_elem)
                         data_out.extend(add_elem)
                     x_pos += x
                 y_pos += y
+
+            self.context.signal("refresh_scene")
             return "elements", data_out
 
+        @context.console_argument("copies", type=int, help=_("Number of copies"))
+        @context.console_argument("radius", type=Length, help=_("Radius"))
+        @context.console_argument("cx", type=Length, help=_("X of center to circle around"))
+        @context.console_argument("cy", type=Length, help=_("Y of center to circle around"))
+        @context.console_argument("startangle", type=Angle.parse, help=_("Start-Angle"))
+        @context.console_argument("endangle", type=Angle.parse, help=_("End-Angle"))
```

Comment:
a) Agree, cy and cy stem from an earlier iteration and make no more sense - removed
b) Not sure I understand what you are missing, I think you can do all with the different parametersets. Especially the deltaangle parameter will just regard the startangle and iterate around the center how long it takes:
<img width="319" alt="image" src="https://user-images.githubusercontent.com/2670784/157449670-3bfc7f1c-84a6-4d5e-a910-dbd2606b1f28.png">

Reply:
Not missing something. I'm pointing out there are other parameters that achieve the same thing. You can parameterize code in a *lot* of different ways especially with regard to circles. `Sweep`, `start`, and `total_copies` gives you the needed parameters. So does `delta` `start` and `total_copies` since delta is just `sweep / total_copies` and you might be able to solve some-things with `start` `end` and `total_copies` but in this case there's an infinite number of different sweep values that could suffice, and 2 of them are going to be less than 1 turn.